util/errors/errors.go

```diff
@@ -61,6 +69,16 @@ func isResourceQuotaTimeoutErr(err error) bool {
 	return apierr.IsInternalError(err) && strings.Contains(err.Error(), "resource quota evaluation timed out")
 }
 
+func isTransientEtcdErr(err error) bool {
+	// Some clusters expose these (transient) etcd errors to the caller
+	if strings.Contains(err.Error(), "etcdserver: leader changed") {
+		return true
+	} else if strings.Contains(err.Error(), "etcdserver: request timed out") {
+		return true
```

Comment:
Can this be merged into isTransientNetworkErr?

Reply:
I did consider it, but it isn't strictly a network error. We're successfully talking to the k8s API, which is telling us to go away because it's not ready right now.

I think the network errors are errors between the workflow-controller (or executor because it also uses this logic) and the k8s API.

That said, I'd rather get this in somewhere than have this block the issue, so if you're sure you want it there I'm happy to change it.